### PR TITLE
[CUMULUS-3075] - Update granule API endpoints to be consistent with empty file objects

### PR DESCRIPTION
**Pull request by Nnaga1**
**Summary:** Summary of changes

Addresses [CUMULUS-3075: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3075)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [x] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: packages/db/src/translate/granules.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export const translatePostgresGranuleToApiGranule = async ({
     endingDateTime: granulePgRecord.ending_date_time?.toISOString(),
     error: granulePgRecord.error,
     execution: executionUrls[0] ? executionUrls[0].url : undefined,
-    files: files.length > 0 ? files.map((file) => translatePostgresFileToApiFile(file)) : undefined,
+    files: files.length > 0 ? files.map((file) => translatePostgresFileToApiFile(file)) : [],
```
Comment (botanical) [inline packages/db/src/translate/granules.ts:98]: 👍

File: packages/api/tests/lib/test-reconciliationReport.js [hunk 2]
```diff
@@ -246,6 +246,7 @@ test('compareEsGranuleAndApiGranule returns true for matching granules', (t) =>
     granuleId: cryptoRandomString({ length: 5 }),
   };
   const granule2 = { ...granule };
+  granule2.files = [];
```
Comment (botanical) [inline packages/api/tests/lib/test-reconciliationReport.js:249]: 🐜 
I think you could also write it this way:
```suggestion
const granule2 = { ...granule, files: [] };
```

File: packages/api/tests/lambdas/test-bulk-operation.js [hunk 3]
```diff
@@ -811,6 +812,7 @@ test.serial('bulk operation BULK_GRANULE_REINGEST reingests granule IDs returned
       knexOrTransaction: knex,
     });
 
+    callArgs[0].granule.files = [];
```
Comment (botanical) [inline packages/api/tests/lambdas/test-bulk-operation.js:815]: Hm I'm having trouble understanding why this change is necessary. Do you mind explaining?

File: packages/api/lib/writeRecords/write-granules.js [hunk 4]
```diff
@@ -859,7 +859,7 @@ const writeGranulesFromMessage = async ({
         s3: s3(),
         providerURL: buildURL(provider),
         files: granule.files,
-      }) : undefined;
+      }) : [];
```
Comment (botanical) [inline packages/api/lib/writeRecords/write-granules.js:862]: 👍

File: packages/api/lib/reconciliationReport.js [hunk 5]
```diff
@@ -263,7 +263,7 @@ function compareEsGranuleAndApiGranule(esGranule, apiGranule) {
   if (granulesAreEqual === false) return granulesAreEqual;
 
   const esGranulesHasFiles = esGranule.files !== undefined;
-  const apiGranuleHasFiles = apiGranule.files !== undefined;
+  const apiGranuleHasFiles = apiGranule.files.length !== 0;
```
Comment (botanical) [inline packages/api/lib/reconciliationReport.js:266]: Should `esGranulesHasFiles` be updated to check for an empty array instead of undefined, as well?

File: CHANGELOG.md [hunk 6]
```diff
@@ -12,6 +12,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Added
 
+-**CUMULUS-3075**
+  - Changed the API endpoint return value for a granule with no files. When a granule has no files, the return value beforehand for
+    the translatePostgresGranuletoApiGranule, the function which does the translation of a Postgres granule to an API granule, was 
+    undefined, now changed to an empty array. 
+  - Existing behavior which relied on the pre-disposed undefined value was changed to instead accept the empty array.
```
Comment (botanical) [inline CHANGELOG.md:19]: There's a section for changes `### Changed` and I think these two points above would be more appropriate there 😮

**Review by botanical — COMMENTED** [review]
Hey, great work @Nnaga1! I had a couple of question on the PR about some test changes and one comment on the reconciliationReport file change.

File: packages/api/tests/lambdas/test-bulk-operation.js (hunk 3 above)
Comment (botanical) [reply packages/api/tests/lambdas/test-bulk-operation.js:813]: I believe `callArgs[0].granule.file` will already be `[]` 🤔

File: CHANGELOG.md (hunk 6 above)
Comment (botanical) [reply CHANGELOG.md]: Hey @Nnaga1! To answer your question here:
`Lastly, would I need to write an integration test or not, I'm not sure what the criteria is for when or when not to do so`, I think generally a good rule is integration tests are appropriate for testing that the changes you made work end-to-end. So if you wanted to, you could add a test to https://github.com/nasa/cumulus/blob/master/example/spec/parallel/testAPI/granuleSpec.js to test that calling the granule API endpoint for a modified granule with an empty files objects returns what you expect.

File: CHANGELOG.md [hunk 7]
```diff
@@ -54,6 +54,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
   - Added configuration to increase the limit for body-parser's JSON and URL encoded parsers to allow for larger input payloads
 
 ### Changed
+- **CUMULUS-3075**
+  - Changed the API endpoint return value for a granule with no files. When a granule has no files, the return value beforehand for
+    the translatePostgresGranuletoApiGranule, the function which does the translation of a Postgres granule to an API granule, was 
+    undefined, now changed to an empty array. 
+  - Existing behavior which relied on the pre-disposed undefined value was changed to instead accept the empty array.
+  - Standardized tests in order to expect an empty array for a granule with no files files' object instead of undefined.
```
Comment (botanical) [inline CHANGELOG.md:62]: This section needs to be moved to `Unreleased`, sorry if I didn't catch this before 😞

**Review by botanical — APPROVED** [review]

File: packages/api/tests/lambdas/test-bulk-operation.js [hunk 8]
```diff
@@ -757,7 +757,8 @@ test.serial('bulk operation BULK_GRANULE_REINGEST reingests list of granule IDs
     t.true(t.context.executionArns.includes(callArgs[0].granule.execution));
     delete matchingGranule.execution;
     delete callArgs[0].granule.execution;
-
+    callArgs[0].granule.files = [];
+    matchingGranule.files = [];
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply packages/api/tests/lambdas/test-bulk-operation.js:760]: It looks like this change is still present?
